### PR TITLE
fix: Popover display empty div when title and content is null

### LINKED ISSUE #42216: Popover 组件当content值为空时，弹出空白
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/ji-ben-antd-5-4-7-forked-ghnwme)

### Steps to reproduce

Popover title 跟 content属性均传空值

### What is expected?

不展示气泡卡片

### What is actually happening?

弹出空白卡片

| Environment | Info |
| --- | --- |
| antd | 4.24.10 |
| React | 18 |
| System | Win10 |
| Browser | Chrome112 |

---
4.x与5.x均有次问题，请务必修复下4.x版本

### PR DESCRIPTION
**Pull request by hairgc**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link
fix #42216 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     fix: Popover display empty div when title and content is null      |
| 🇨🇳 Chinese |    修复: 当title和content属性均为空值时，Popover组件展示空白气泡      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at eab7c90</samp>

Simplify `Popover` component logic and rendering. Remove redundant `title` and `content` props from `OverlayProps` and render `Overlay` only when it has content.

### <samp>🤖 Generated by Copilot at eab7c90</samp>

* Remove unnecessary check for `title` and `content` in `OverlayProps` interface ([link](https://github.com/ant-design/ant-design/pull/42217/files?diff=unified&w=0#diff-8255c20b0fc82fa28472ee578129241aab0b6c6d3f2cdafba5f74fa2614a92f0L25-L27))
* Conditionally render `Overlay` component in `Popover` component based on `title` or `content` ([link](https://github.com/ant-design/ant-design/pull/42217/files?diff=unified&w=0#diff-8255c20b0fc82fa28472ee578129241aab0b6c6d3f2cdafba5f74fa2614a92f0L68-R67))

### CODE REVIEW

**Comment by hairgc** [comment]
请同步修复下4.x版本

**Comment by yoyo837** [comment]
> 请同步修复下4.x版本

再给4.x-stable分支发PR即可

File: components/popover/index.tsx [hunk 1]
```diff
@@ -22,9 +22,6 @@ interface OverlayProps {
 }
 
 const Overlay: React.FC<OverlayProps> = ({ title, content, prefixCls }) => {
-  if (!title && !content) {
-    return null;
-  }
   return (
     <>
```
Comment (yoyo837) [inline components/popover/index.tsx:26]: `Fragment` 是不是可以干掉

**Comment by zombieJ** [comment]
加一个 test case~

File: components/popover/index.tsx (hunk 1 above)
Comment (MadCcc) [reply components/popover/index.tsx:26]: 不行吧，下面两个节点

File: components/popover/index.tsx [hunk 2]
```diff
@@ -65,7 +62,9 @@ const Popover = React.forwardRef<unknown, PopoverProps>((props, ref) => {
       prefixCls={prefixCls}
       overlayClassName={overlayCls}
       ref={ref}
-      overlay={<Overlay prefixCls={prefixCls} title={title} content={content} />}
+      overlay={
```
Comment (li-jia-nan) [inline components/popover/index.tsx:65]: 建议用三元，用 null 作为空节点

**Comment by li-jia-nan** [comment]
同步一下上游分支

File: components/popover/index.tsx (hunk 1 above)
Comment (afc163) [reply components/popover/index.tsx:26]: 看下 CodeFactor 的提示

File: components/popover/index.tsx [hunk 3]
```diff
@@ -22,9 +22,6 @@ interface OverlayProps {
 }
 
 const Overlay: React.FC<OverlayProps> = ({ title, content, prefixCls }) => {
-  if (!title && !content) {
```
Comment (afc163) [inline components/popover/index.tsx:25]: 看下 CodeFactor 的提示

File: components/popover/index.tsx (hunk 2 above)
Comment (hairgc) [reply components/popover/index.tsx:65]: ok

**Comment by hairgc** [comment]
> 加一个 test case~

已加，原测试案例写的有点问题，顺带修复了

File: components/popover/index.tsx (hunk 3 above)
Comment (li-jia-nan) [reply components/popover/index.tsx:25]: 这里把 `{}` 和 `return` 省略掉就好了

**Comment by hairgc** [comment]
> > 请同步修复下4.x版本
> 
> 再给4.x-stable分支发PR即可

等这个review无问题后，我这边再发起一个到4.x的PR

File: components/popover/index.tsx (hunk 1 above)
Comment (yoyo837) [reply components/popover/index.tsx]: > 不行吧，下面两个节点

OK, 没注意到.

File: components/popover/index.tsx (hunk 3 above)
Comment (hairgc) [reply components/popover/index.tsx:25]: 已改

**Review by li-jia-nan — APPROVED** [review]

**Pull request merged**